datafusion/expr/src/logical_plan/ddl.rs

```diff
@@ -303,3 +321,66 @@ pub struct DropCatalogSchema {
     /// Dummy schema
     pub schema: DFSchemaRef,
 }
+
+#[derive(Clone, PartialEq, Eq, Hash, Debug)]
+pub struct CreateFunction {
+    // TODO: There is open question should we expose sqlparser types or redefine them here?
```

Comment:
Is this still TODO (or are the rest of the comments below todo)? I think this structure makes sense and with a little more explanation on rationale (aka the Statement::CreateFunction is a variant and not its own struct) this make sense

The rationale I think would be important so that when we upgrade sqlparser's version and it adds more CREATE FUNCTION variants, the updater knows the desired results in DataFusion